src/lib/mina_base/parties_logic.ml

```diff
@@ -1095,6 +1108,24 @@ module Make (Inputs : Inputs_intf) = struct
       let a = Account.set_delegate delegate a in
       (a, local_state)
     in
+    (* Update nonce. *)
+    let a, local_state =
+      let nonce = Account.nonce a in
+      let increment_nonce = Party.increment_nonce party in
+      let nonce =
+        Nonce.if_ increment_nonce ~then_:(Nonce.succ nonce) ~else_:nonce
+      in
+      let has_permission =
+        Controller.check ~proof_verifies ~signature_verifies
+          (Account.Permissions.increment_nonce a)
+      in
+      let local_state =
+        Local_state.add_check local_state Update_not_permitted_delegate
```

Comment:
change error to `Update_not_permitted_nonce`

Reply:
Good catch!